bigbone/src/integrationTest/kotlin/social/bigbone/v300/V300StatusMethodsIntegrationTest.kt

```diff
@@ -15,6 +15,7 @@ import social.bigbone.api.entity.Token
  */
 @TestInstance(TestInstance.Lifecycle.PER_CLASS)
 class V300StatusMethodsIntegrationTest {
+    private val fullScope = Scope(Scope.Name.READ, Scope.Name.WRITE, Scope.Name.PUSH)
```

Comment:
We define this one in multiple places now. How about moving the definition to a `companion object` inside `Scope`?


Could look like this then and would be referenced in classes like this test here as `Scope.FULL_SCOPE`:

```kotlin
import social.bigbone.api.Scope.Name.PUSH
import social.bigbone.api.Scope.Name.READ
import social.bigbone.api.Scope.Name.WRITE

companion object {
    val FULL_SCOPE = Scope(READ, WRITE, PUSH)
}
```

Reply:
I'm not sure about that, because we would again expose some "shortcut" value to request all available scopes at once when that is not the recommended way, and which on top of that might lead to problems again when the definition of what constitutes a "full scope" changes.

I replaced `Scope.ALL` with the more verbose `Scope(READ, WRITE, PUSH)` because that was the minimally invasive way. If the repeated definition of a "full scope" is problematic, I'd rather do the following:
1. in unit tests, remove the `scope` parameter completely, because it has a default value and isn't testable there anyway
2. for the integration tests, move this definition into the existing `TestConstants` class
3. for our samples, enhance them by allowing to get different access tokens with different scopes, and clarifying in each sample which scope(s) will be necessary for them to run successfully

I can do 1 and 2 as part of this PR, but would prefer to do 3 in a separate PR for clarity.

Reply:
Okay, I get that. 👍 
1+2 sound like a good idea!

I agree with 3 being too much here. We might also want to update our kDoc to contain the scope expected for the methods we expose—currently we do not do that at all.

Reply:
1 and 2 are done. Will create a separate issue for 3 - and also for the kdoc enhancement, good idea!